### ISSUE TITLE
performans metrics

### ISSUE THREAD

**Issue by idrisdemir** [zhenghuizhao/TransWCD#5]
Hello,

I recently ran your code on the LEVIR-CD dataset and obtained class-specific evaluation metrics such as F1-score and precision. However, in the paper, only a single value is reported for each metric, which left me a bit confused. For instance, I got precision values like {0: 0.98..., 1: 0.43}, while the paper reports a precision of 55.48 for TransWCD.

I used the original code without modifying the parameters, except for making adjustments to run it on my local machine. The config file had the number of steps set to 50,000, so I trained the model accordingly. Later, while reviewing the paper again, I noticed that it mentions training for 30,000 steps. So I checked the logs, found the best-performing weights around step 30,000, and evaluated the model using those weights as well.

I’d really appreciate your guidance in understanding how to correctly interpret or reproduce the results from the paper. Thank you in advance for your help!

**Comment by zhenghuizhao** [comment]
Apologies for the delay, I’ve been quite busy recently. Thank you for your attention! In the precision values {0: 0.98..., 1: 0.43}, 0 refers to the unchanged category, and 1 refers to the changed category. The F1 score reported in the paper corresponds to the changed category. It is recommended to use a larger batch size, and by following the settings in  https://github.com/zhenghuizhao/TransWCD/tree/main/transwcd/configs for each dataset, you should be able to reproduce the results.

**Comment by idrisdemir** [comment]
Thank you for your response. I increased the batch_size from 8 to 64 and used the LEVIR-CD dataset. I obtained the test results using the best-performing model. As a result, the precision improved slightly from 0.43 to 0.44. I have also attached the relevant files for your reference. I would appreciate it if you could kindly check whether there might be any mistakes or issues in my implementation.

[test.txt](https://github.com/user-attachments/files/21543352/test.txt)

[train_transwcd.txt](https://github.com/user-attachments/files/21543353/train_transwcd.txt)

[LEVIR.txt](https://github.com/user-attachments/files/21543356/LEVIR.txt)

Since GitHub does not allow me to upload files with .py and .yaml extensions, I uploaded them as .txt files instead.

**Comment by zhenghuizhao** [comment]
You could try modifying the config file by setting cam: scales: [1, 0.5, 1.5, 2.0] while keeping the batch size at 8. This scale augmentation may help improve the CAM quality. Please ensure that the data loading and augmentation strategy remains unchanged.

Feel free to let me know if you encounter any other issues.

**Comment by idrisdemir** [comment]
Thanks for the suggestions. I tried the parameter settings you mentioned, but the precision is approximately 0.45. Let me know if you think there's something else I should try.

**Comment by zhenghuizhao** [comment]
You are welcome! Maybe you can try adding 
“self.dropout = nn.Dropout2d(0.1) 
_c4 = self.dropout(_c4)” at [https://github.com/zhenghuizhao/TransWCD/tree/main/transwcd/models/model_transwcd.py]

**Comment by idrisdemir** [comment]
I have downloaded your best model and attempted to run the testing process. Using the LEVIR-CD dataset, the reported F1 score for TransWCD-Dual is 62.55, but I obtained 62.41. With the WHU-CD dataset, your reported F1 score is 68.73, whereas I achieved 63.98.

I carefully installed all the libraries and versions listed in the requirements.txt file and used CUDA 11.1. I understand that your setup was likely based on Python 3.6, but I used Python 3.8 as mmcv=2.0.1 is not compatible with Python 3.6. Even when using the provided model and running only test.py, I was unable to fully reproduce the reported results.

I would greatly appreciate any guidance on potential reasons for this difference. Additionally, I noticed that the LEVIR.yaml file does not include an ignore_index parameter, although test.py attempts to access it. Since the code appears to have a default value, I commented out that line.

Thank you very much for your time and support.

**Comment by zhenghuizhao** [comment]
Thank you for your effort in testing and for pointing this out. Your reproduced result on LEVIR-CD (62.41 vs. our 62.55) is almost identical to ours, which shows that the model is working correctly. It’s quite possible that the problem of WHU-CD arises from the data difference. As WHU-CD does not provide a predefined split, the partitioning we used for training and evaluation may differ from yours, which could explain the discrepancy in the reported results.
